intro.js

```diff
@@ -136,34 +137,40 @@
           currentItem.disableInteraction = this._options.disableInteraction;
         }
 
-        if (currentItem.element != null) {
+        if (currentItem.element !== null) {
           introItems.push(currentItem);
         }
       }
 
     } else {
       //use steps from data-* annotations
       var allIntroSteps = targetElm.querySelectorAll('*[data-intro]');
+      var elmsLength = allIntroSteps.length;
+      var currentElement;
+      var disableInteraction;
+
       //if there's no element to intro
-      if (allIntroSteps.length < 1) {
+      if (elmsLength < 1) {
         return false;
       }
 
+      i = 0;
+
       //first add intro items with data-step
-      for (var i = 0, elmsLength = allIntroSteps.length; i < elmsLength; i++) {
-        var currentElement = allIntroSteps[i];
+      for (; i < elmsLength; i++) {
```

Comment:
I think it makes more sense to define `i` inside the for (statement 1)

Reply:
I agree, there might be an option in jshint to change how it interprets this.  Basically, you can't have two forloops in the same block-scope re-defining the same variables. So:

```javascript
for (var i = 0...) {} 
// and then
for (var i = 0...) {}
```

Is re-defining `i`, which jshint disagrees with, and though I think it's good practice, I also agree with you that it makes things look a little more awkward (though maybe there's just a better way to do it).  ES6 recommends using `let` instead of `var`, and we can start doing that in a later version.  

I suppose for this we could wrap it in a closure? Or we can look into changing the jshint options.  What do you think?

Reply:
And actually maybe the solution here is just to remove the `var`:

```javascript
for (i = 0, i < elmsLength; i++)
```

That probably solves it and makes it just as readable